src/coprocessor/codec/mysql/time.rs

```diff
@@ -311,6 +328,41 @@ impl Time {
         Time::new(t, tp, fsp as i8)
     }
 
+    pub fn from_duration(tz: &FixedOffset, d: &MyDuration) -> Result<Time> {
+        let dur = Duration::nanoseconds(d.to_nanos());
+        let t = Utc::now()
```

Comment:
Does it mean that when the query run across midnight, we will get inconsistent result?